src/lint/track_config.nim

```diff
@@ -238,6 +240,36 @@ type
     exercises: Exercises
     concepts: Concepts
 
+proc toLineAndCol(s: string; offset: Natural): tuple[line: int; col: int] =
+  ## Returns the line and column number corresponding to the `offset` in `s`.
+  result = (1, 1)
+  for i, c in s:
+    if i == offset:
+      break
+    elif c == '\n':
+      inc result.line
+      result.col = 0
+    inc result.col
+
+proc toTrackConfig(trackConfigContents: string): TrackConfig =
+  ## Deserializes `trackConfigContents` using `jsony` to a `TrackConfig` object.
+  try:
+    result = fromJson(trackConfigContents, TrackConfig)
+  except jsony.JsonError:
+    let jsonyMsg = getCurrentExceptionMsg()
+    var msg = "JSON parsing error during the second linting pass:\nconfig.json"
```

Comment:
I would argue that configlet users don't have to know about first/second passes, but just about errors:

```suggestion
    var msg = "JSON parsing error:\nconfig.json"
```

Reply:
I wrote it like this to make it more distinct from the other JSON parsing errors, on the basis that:
1. This error indicates a bug in `configlet`, rather than a problem with `config.json`, because type errors in `config.json` should be caught by the first pass.
2. We build the release binary with `-d:release`, so there's no stacktrace for errors

But if a user opens an issue for the error message, I think we'll be able to detect that it happens during the second pass anyway. So I've pushed a commit with your suggestion (on top of some refactoring).

But now that I think about it, given point 1 above, I'm leaning towards adding something like:

> `If you see this error message, please open an issue in https://github.com/exercism/configlet, including the contents of the track config.json file  at the time configlet lint ran`

What do you think?

Reply:
I like that suggestion. It should indeed not happen

Reply:
Pushed.

Visible enough? :)